### PR TITLE
Switch pymacaroons to use pynacl not libnacl

### PR DESCRIPTION
**Pull request by illicitonion**
I accidentally branched this off of the base of PR3. Oops.

### CODE REVIEW

**Comment by NegativeMjark** [comment]
Have you tried running "tox" over this?

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Actual Dependencies
 -e .
+pynacl>0.0.3
```
Comment (NegativeMjark) [inline requirements.txt]: I don't think this is necessary if you're adding pynacl to the setup.py dependencies.

File: pymacaroons/caveat_delegates/third_party.py [hunk 2]
```diff
@@ -27,14 +28,16 @@ def add_third_party_caveat(self,
                                location,
                                key,
                                key_id,
+                               nonce = None,
```
Comment (NegativeMjark) [inline pymacaroons/caveat_delegates/third_party.py]: flake8: Space around keyword argument

File: pymacaroons/field_encryptors/secret_box_encryptor.py [hunk 3]
```diff
@@ -16,7 +18,8 @@ def __init__(self, signifier=None, nonce=None):
         super(SecretBoxEncryptor, self).__init__(
             signifier=signifier or 'sbe::'
         )
-        self.nonce = nonce
+        self.nonce = (nonce or
+            nacl.utils.random(nacl.bindings.crypto_secretbox_NONCEBYTES))
```
Comment (NegativeMjark) [inline pymacaroons/field_encryptors/secret_box_encryptor.py]: I and flake8 would prefer something like:

```
self.nonce = nonce or nacl.utils.random(
   nacl.bindings.crypto_secretbox_NONCEBYTES
)
```

File: requirements.txt (hunk 1 above)
Comment (illicitonion) [inline requirements.txt]: Do you mean this change, or this whole file?

File: pymacaroons/caveat_delegates/third_party.py (hunk 2 above)
Comment (illicitonion) [inline pymacaroons/caveat_delegates/third_party.py]: Done

File: pymacaroons/field_encryptors/secret_box_encryptor.py (hunk 3 above)
Comment (illicitonion) [inline pymacaroons/field_encryptors/secret_box_encryptor.py]: Bleh, I prefer breaking on higher order concepts if they can break cleanly. But this is a fight we can have another day :) Done

**Comment by illicitonion** [comment]
I have had no joy getting tox running, but nose, which tox is trying to run, is happy:
$ nosetests-2.7 $(find tests -name "_.py")
/home/wadaniel/src/pymacaroons/tests/property_tests/macaroon_property_tests.py:14: HypothesisDeprecationWarning: Calling strategy with non-strategy object [SampledFrom(elements=('\x00', '\x01', '\x02', '\x03', '\x04', '\x05', '\x06', '\x07', '\x08', '\t', '\n', '\x0b', '\x0c', '\r', '\x0e', '\x0f', '\x10', '\x11', '\x12', '\x13', '\x14', '\x15', '\x16', '\x17', '\x18', '\x19', '\x1a', '\x1b', '\x1c', '\x1d', '\x1e', '\x1f', ' ', '!', '"', '#', '$', '%', '&', "'", '(', ')', '_', '+', ',', '-', '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', ':', ';', '<', '=', '>', '?', '@', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '[', '\', ']', '^', '_', '`', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '{', '|', '}', '~', '\x7f'))] is deprecated and will be removed in Hypothesis 2.0. Use the functions in hypothesis.strategies instead.
  [sampled_from(map(chr, range(0, 128)))]
/home/wadaniel/.virtualenvs/pymacaroons/local/lib/python2.7/site-packages/hypothesis/strategies.py:830: HypothesisDeprecationWarning: Calling strategy with non-strategy object SampledFrom(elements=('\x00', '\x01', '\x02', '\x03', '\x04', '\x05', '\x06', '\x07', '\x08', '\t', '\n', '\x0b', '\x0c', '\r', '\x0e', '\x0f', '\x10', '\x11', '\x12', '\x13', '\x14', '\x15', '\x16', '\x17', '\x18', '\x19', '\x1a', '\x1b', '\x1c', '\x1d', '\x1e', '\x1f', ' ', '!', '"', '#', '$', '%', '&', "'", '(', ')', '*', '+', ',', '-', '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', ':', ';', '<', '=', '>', '?', '@', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '[', '\', ']', '^', '_', '`', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '{', '|', '}', '~', '\x7f')) is deprecated and will be removed in Hypothesis 2.0. Use the functions in hypothesis.strategies instead.
  return lists(one_of(*[strategy(s, settings) for s in specifier]))
...................../home/wadaniel/.virtualenvs/pymacaroons/local/lib/python2.7/site-packages/hypothesis/core.py:504: HypothesisDeprecationWarning: Calling strategy with non-strategy object OneOf(elements=(lists(elements=sampled_from(('\x00', '\x01', '\x02', '\x03', '\x04', '\x05', '\x06', '\x07', '\x08', '\t', '\n', '\x0b', '\x0c', '\r', '\x0e', '\x0f', '\x10', '\x11', '\x12', '\x13', '\x14', '\x15', '\x16', '\x17', '\x18', '\x19', '\x1a', '\x1b', '\x1c', '\x1d', '\x1e', '\x1f', ' ', '!', '"', '#', '$', '%', '&', "'", '(', ')', '*', '+', ',', '-', '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', ':', ';', '<', '=', '>', '?', '@', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '[', '\\', ']', '^', '_', '`', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '{', '|', '}', '~', '\x7f'))).map(lambda c: <unknown>), lists(elements=sampled_from(('\x00', '\x01', '\x02', '\x03', '\x04', '\x05', '\x06', '\x07', '\x08', '\t', '\n', '\x0b', '\x0c', '\r', '\x0e', '\x0f', '\x10', '\x11', '\x12', '\x13', '\x14', '\x15', '\x16', '\x17', '\x18', '\x19', '\x1a', '\x1b', '\x1c', '\x1d', '\x1e', '\x1f', ' ', '!', '"', '#', '$', '%', '&', "'", '(', ')', '*', '+', ',', '-', '.', '/', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', ':', ';', '<', '=', '>', '?', '@', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '[', '\', ']', '^', '_', '`', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '{', '|', '}', '~', '\x7f'))).map(lambda c: <unknown>).map(lambda s: convert_to_bytes(s)))) is deprecated and will be removed in Hypothesis 2.0. Use the functions in hypothesis.strategies instead.
  return strategy(v.value, settings)

## .

Ran 22 tests in 0.725s

OK

File: requirements.txt (hunk 1 above)
Comment (NegativeMjark) [inline requirements.txt]: This change.

Comment (illicitonion) [inline requirements.txt]: Done

**Comment by NegativeMjark** [comment]
LGTM

**Pull request merged**